### PR TITLE
Add tests for book and category entity

### PR DESCRIPTION
**Pull request by yulka-tsybulka**

### CODE REVIEW

**Comment by yulka-tsybulka** [comment]
Tests coverage

File: src/test/java/my/book/market/config/CustomMySqlContainer.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        super.start();
+        System.setProperty("TEST_DB_URL", mySqlContainer.getJdbcUrl());
+        System.setProperty("TEST_DB_USERNAME", mySqlContainer.getUsername());
+        System.setProperty("TEST_DB_PASSWORD", mySqlContainer.getPassword());
+    }
+
+    @Override
+    public void stop() {
+        super.stop();
+    }
```
Comment (koval-kn) [inline src/test/java/my/book/market/config/CustomMySqlContainer.java]: changes nothing
```suggestion
```

File: src/test/java/my/book/market/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CategoryControllerTest {
```
Comment (koval-kn) [inline src/test/java/my/book/market/controller/CategoryControllerTest.java:41]: add negative tests

File: src/test/java/my/book/market/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
```
Comment (koval-kn) [inline src/test/java/my/book/market/controller/BookControllerTest.java:45]: add negative tests

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

**Pull request merged**